drivers/usb/device/usb_dc_stm32.c

```diff
@@ -127,14 +127,18 @@ static const struct gpio_dt_spec ulpi_reset =
 #define EP_MPS USB_OTG_FS_MAX_PACKET_SIZE
 #endif
 
-/* We need one RX FIFO and n TX-IN FIFOs */
-#define FIFO_NUM (1 + USB_NUM_BIDIR_ENDPOINTS)
+/* We need n TX IN FIFOs */
+#define TX_FIFO_NUM USB_NUM_BIDIR_ENDPOINTS
 
-/* 4-byte words FIFO */
-#define FIFO_WORDS (USB_RAM_SIZE / 4)
+/* FIFO Rx Minimum memory to works */
+#define FIFO_RX_MIN 160
 
-/* Allocate FIFO memory evenly between the FIFOs */
-#define FIFO_EP_WORDS (FIFO_WORDS / FIFO_NUM)
+/* 4-byte words TX FIFO */
+#define TX_FIFO_WORDS ((USB_RAM_SIZE - FIFO_RX_MIN) / 4)
```

Comment:
Shouldn't we also take into account the 64 bytes assigned to EP0 in this ?

Reply:
yes sorry